executor/hash_table.go

```diff
@@ -47,8 +48,28 @@ const (
 type hashContext struct {
 	allTypes  []*types.FieldType
 	keyColIdx []int
-	h         hash.Hash64
 	buf       []byte
+	hashVals  []hash.Hash64
+	hasNull   []bool
+}
+
+func (hc *hashContext) initHash(rows int) {
+	if hc.buf == nil {
+		hc.buf = make([]byte, 1)
+	}
+
+	if len(hc.hashVals) < rows {
+		hc.hasNull = make([]bool, rows)
+		hc.hashVals = make([]hash.Hash64, rows)
+		for i := 0; i < rows; i++ {
+			hc.hashVals[i] = fnv.New64()
+		}
+	} else {
+		for i := 0; i < rows; i++ {
```

Comment:
It will be nice to
``` go
hc.hasNull = hc.hasNull[:rows]
hc.hashVals = hc.hashVals[:rows]
```

Reply:
This will leave the values unchanged.
We need to reset the values before every run.